### PR TITLE
[SPARK-49310][BUILD] Upgrade `Parquet` to 1.14.2

### PR DESCRIPTION
**Pull request by Fokko**
### What changes were proposed in this pull request?

This PR aims to upgrade Parquet to 1.14.2.

### Why are the changes needed?

To bring the latest bug fixes.
- https://mvnrepository.com/artifact/org.apache.parquet/parquet-common/1.14.2

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Yay. Thank you, @Fokko !

File: pom.xml [hunk 1]
```diff
@@ -348,6 +348,17 @@
         <enabled>false</enabled>
       </snapshots>
     </repository>
+    <repository>
+      <id>parquet-release-candidate</id>
+      <name>Parquet Release Candidate</name>
+      <url>https://repository.apache.org/content/repositories/orgapacheparquet-1058</url>
```
Comment (Fokko) [inline pom.xml:354]:
```suggestion
      <url>https://repository.apache.org/content/repositories/orgapacheparquet-1059</url>
```

File: project/SparkBuild.scala [hunk 2]
```diff
@@ -275,7 +275,8 @@ object SparkBuild extends PomBuild {
       "gcs-maven-central-mirror" at "https://maven-central.storage-download.googleapis.com/maven2/",
       DefaultMavenRepository,
       Resolver.mavenLocal,
-      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
+      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns),
+      "parquet-release-candidate" at "https://repository.apache.org/content/repositories/orgapacheparquet-1058"
```
Comment (Fokko) [inline project/SparkBuild.scala:279]:
```suggestion
      "parquet-release-candidate" at "https://repository.apache.org/content/repositories/orgapacheparquet-1059"
```

**Review by dongjoon-hyun — COMMENTED** [review]
Could you update this PR, @Fokko ? 😄

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs).

To @Fokko , I revised your PR code and the description a little.

I hope we can deliver this with Apache Spark 4.0.0-preview2 (RC1) next Monday.

Please let me know if you have any concern on Apache Parquet 1.14.2.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 4.0.0-preview2.

**Pull request closed without merging**